spec/factorial_spec.rb

```diff
@@ -0,0 +1,35 @@
+require_relative '../main'
+
+describe Solver do
+  before(:each) do
+    @solver = Solver.new
+  end
```

Comment:
- [ ] Please make sure that you have a test case that factorial of Negative value raises an exception as an expectation. :pray: